Emby.Server.Implementations/Library/Resolvers/BaseVideoResolver.cs

```diff
@@ -156,19 +157,26 @@ protected void SetIsoType(Video video)
                 }
                 else
                 {
-                    // use disc-utils, both DVDs and BDs use UDF filesystem
-                    using (var videoFileStream = File.Open(video.Path, FileMode.Open, FileAccess.Read))
-                    using (UdfReader udfReader = new UdfReader(videoFileStream))
+                    try
                     {
-                        if (udfReader.DirectoryExists("VIDEO_TS"))
+                        // use disc-utils, both DVDs and BDs use UDF filesystem
+                        using (var videoFileStream = File.Open(video.Path, FileMode.Open, FileAccess.Read))
+                        using (UdfReader udfReader = new UdfReader(videoFileStream))
                         {
-                            video.IsoType = IsoType.Dvd;
-                        }
-                        else if (udfReader.DirectoryExists("BDMV"))
-                        {
-                            video.IsoType = IsoType.BluRay;
+                            if (udfReader.DirectoryExists("VIDEO_TS"))
+                            {
+                                video.IsoType = IsoType.Dvd;
+                            }
+                            else if (udfReader.DirectoryExists("BDMV"))
+                            {
+                                video.IsoType = IsoType.BluRay;
+                            }
                         }
                     }
+                    catch (Exception ex)
+                    {
+                        Video.Logger.LogError(ex, "Error opening UDF/ISO image: {Value}", video.Path ?? video.Name);
```

Comment:
Video.Logger is as legacy as it gets. The logger should be injected into the constructor.

Reply:
The only way I see is to propagate if all the way from the `LibraryManager`.
Any better suggestions?

Reply:
Propagate it? The resolver classes are created by the DI container. It's just a matter of adding it to the constructor

Reply:
> Propagate it? The resolver classes are created by the DI container. It's just a matter of adding it to the constructor

[The `LibraryManager` explicitly instantiates the `ExtraResolver` -> [`GenericResolver<Trailer>` -> `GenericResolver<Video>`, `<VideoResolver>` -> `BaseVideoResolver`].](https://github.com/jellyfin/jellyfin/blob/ce61dff4aae0875cfc359c9d8dc1a8a15f9409cd/Emby.Server.Implementations/Library/LibraryManager.cs#L150)



I switched it to `ActivatorUtilities.CreateInstance` to be able to inject the `ILogger` into all `Resolver`s created that way.